addons/mrp/models/stock_orderpoint.py

```diff
@@ -49,13 +51,55 @@ def _compute_allowed_replenishment_uom_ids(self):
             if 'manufacture' in orderpoint.rule_ids.mapped('action'):
                 orderpoint.allowed_replenishment_uom_ids += orderpoint.product_id.bom_ids.product_uom_id
 
-    @api.depends('route_id')
+    @api.onchange('bom_id')
+    def _set_new_route_from_bom(self):
+        for orderpoint in self:
+            if not orderpoint.route_id and orderpoint.bom_id:
+                orderpoint.route_id = self.env['stock.rule'].search([('action', '=', 'manufacture')])[0].route_id
+
+    @api.onchange('route_id')
+    def _clear_bom_if_no_route(self):
+        for orderpoint in self:
+            if not orderpoint.route_id:
+                orderpoint.bom_id = False
+
+    @api.depends('route_id', 'route_id_placeholder')
     def _compute_show_bom(self):
         manufacture_route = []
         for res in self.env['stock.rule'].search_read([('action', '=', 'manufacture')], ['route_id']):
             manufacture_route.append(res['route_id'][0])
         for orderpoint in self:
-            orderpoint.show_bom = orderpoint.route_id.id in manufacture_route
+            orderpoint.show_bom = (orderpoint.route_id.id in manufacture_route) or (orderpoint._get_default_route().id in manufacture_route)
+
+    @api.depends('route_id', 'route_id_placeholder')
+    def _compute_bom_id_placeholder(self):
+        for orderpoint in self:
+            default_bom = orderpoint._get_default_bom()
+            orderpoint.bom_id_placeholder = default_bom.display_name if default_bom else ''
+
+    def _get_default_bom(self):
```

Comment:
follow the guidelines with the order of function https://www.odoo.com/documentation/18.0/contributing/development/coding_guidelines.html#symbols-and-conventions